src/scripts/main.js

```diff
@@ -1,5 +1,35 @@
 'use strict';
 
 document.addEventListener('click', (e) => {
-  // write code here
+  const spider = document.querySelector('.spider');
+  const wall = document.querySelector('.wall');
+  const wallRect = wall.getBoundingClientRect();
+
+  if (!e.target.closest('.wall')) {
+    return;
+  }
+
+  let coordX =
+    e.clientX - wall.clientLeft - wallRect.left - spider.clientWidth / 2;
+  let coordY =
+    e.clientY - wallRect.y - wall.clientTop - spider.clientHeight / 2;
+
+  if (coordX < 0) {
```

Comment:
Consider encapsulating the boundary checks for `coordX` and `coordY` into a separate function to improve code organization and readability. This function can be named something like `ensureWithinBounds` and can be reused for both X and Y coordinate checks.